### PR TITLE
Enable slash-escape in regexp

### PR DESCRIPTION
**Pull request by vruello**

### CODE REVIEW

File: ivre/utils.py [hunk 1]
```diff
@@ -305,20 +305,50 @@ def get_domains(name):
     return ('.'.join(name[i:]) for i in range(len(name)))
 
 
+def espace_slash(string):
+    """This function transforms '\/' in '/'' if '\' was not escaped
```
Comment (p-l-) [inline ivre/utils.py]:
```suggestion
    """This function transforms '\\/' in '/' but leaves '\\\\/' unchanged. This is
useful to parse regexps from JavaScript style (/regexp/).
```

File: ivre/utils.py [hunk 2]
```diff
@@ -305,20 +305,50 @@ def get_domains(name):
...
+    """  # noqa: W605
```
Comment (p-l-) [inline ivre/utils.py]:
```suggestion
    """
```

File: ivre/utils.py [hunk 3]
```diff
@@ -305,20 +305,50 @@ def get_domains(name):
     return ('.'.join(name[i:]) for i in range(len(name)))
 
 
+def espace_slash(string):
```
Comment (p-l-) [inline ivre/utils.py]:
```suggestion
def _escape_slash(string):
```

File: ivre/utils.py [hunk 4]
```diff
@@ -305,20 +305,50 @@ def get_domains(name):
...
 def str2regexp(string):
     """This function takes a string and returns either this string or
     a python regexp object, when the string is using the syntax
     /regexp[/flags].
-
     """
     if string.startswith('/'):
         string = string[1:].rsplit('/', 1)
+        # Enable slash-escape even if it is not necessary
+        string[0] = espace_slash(string[0])
```
Comment (p-l-) [inline ivre/utils.py]:
```suggestion
        string[0] = _escape_slash(string[0])
```

File: ivre/utils.py [hunk 5]
```diff
@@ -305,20 +305,50 @@ def get_domains(name):
...
+        elif escaping and char != '/':
+            new_string += '\\' + char
+            escaping = False
+        else:
+            new_string += char
+            escaping = False
+    return new_string
+
+
+def escape_first_slash(string):
```
Comment (p-l-) [inline ivre/utils.py]: I don't get the purpose of this function.

File: ivre/web/utils.py [hunk 6]
```diff
@@ -138,7 +137,7 @@ def query_from_params(params):
             else [neg, pval, None]
             for neg, pval in (
                 (True, x[1:]) if x[:1] in '!-' else (False, x)
-                for x in shlex.split(query)
+                for x in query.split()
```
Comment (p-l-) [inline ivre/web/utils.py]: Is this related to this PR?

File: ivre/utils.py (hunk 5 above)
Comment (vruello) [reply ivre/utils.py]: If one wants to search for the string `/test`, one would use `\/test` so that the first `/` does not make the string interpreted as a regular expression. Then, it is needed to remove this extra `\`.

File: ivre/web/utils.py (hunk 6 above)
Comment (vruello) [reply ivre/web/utils.py]: Yes! `shlex` causes strange behaviors in web filters. For example, if one wants to match the regex `\d*`, one would use `<something>:/\d*/`. `shlex.split()` interprets the given string as a "shell-like" command and transforms `\d` in `d`. Thus, the pattern checked is `/d*/` which has a significantly different meaning. Currently, the workaround is to escape `\` like `/\\d*/`, which does not seem user friendly to me.

Comment (p-l-) [reply ivre/web/utils.py]: I get it. The problem here is that `shlex.split()` is quite different than `str.split()`:
```
In [1]: "m1 'm2 m3'".split()                                                                                                                                   
Out[1]: ['m1', "'m2", "m3'"]

In [2]: import shlex                                                                                                                                           

In [3]: shlex.split("m1 'm2 m3'")                                                                                                                              
Out[3]: ['m1', 'm2 m3']
```

Comment (vruello) [reply ivre/web/utils.py]: You're right! Perhaps the right solution would be to escape all `\` automatically before calling `shlex.split` : 
```python
     try:
        query = query.replace('\\', '\\\\')
        return [
            [neg] + pval.split(':', 1) if ':' in pval
            else [neg, pval, None]
            for neg, pval in (
                (True, x[1:]) if x[:1] in '!-' else (False, x)
                for x in shlex.split(query)
            )
        ]
```
Does it feel right to you?

Comment (p-l-) [reply ivre/web/utils.py]: Could be. We have to think about this and check if we are not forgetting other corner cases.

Comment (vruello) [reply ivre/web/utils.py]: I'm going to update the code. We'll see if one of the tests fails (every corner cases are not tested, but still it's a first step).
